internal/go_repository.bzl

```diff
@@ -110,6 +110,8 @@ def _go_repository_impl(ctx):
     env.update({k: ctx.os.environ[k] for k in env_keys if k in ctx.os.environ})
 
     if fetch_repo_args:
+        # Override external GO111MODULE, because it is needed by module mode, no-op in repository mode
+        fetch_repo_env["GO111MODULE"] = "on"
```

Comment:
This line needs to be after L117, since `fetch_repo_env` hasn't been defined yet.